packages/react/src/components/cookieConsent/modal/Modal.tsx

```diff
@@ -19,32 +19,63 @@ export function Modal(): React.ReactElement | null {
   // if hasUserHandledAllConsents() was false at first and then later true, user must have saved consents.
   const showScreenReaderSaveNotification = isModalInitiallyShown && !shouldShowModal;
   const { settingsSaved } = useCookieConsentUiTexts();
-  const containerId = 'HdsCookieConsentContainer';
+  const containerId = 'HdsCookieConsentModalContainer';
+  const placeholderId = 'HdsCookieConsentModalPlaceholder';
   const getContainerElement = (): HTMLElement | null => document.getElementById(containerId);
+  const getPlaceholderElement = (): HTMLElement | null => document.getElementById(placeholderId);
   const [isDomReady, setIsDomReady] = useState<boolean>(false);
+  const [contentHeight, setContentHeight] = useState<number>(0);
+  const modalContentRef = useRef<HTMLDivElement>();
+  const bodyBottomPaddingStyleRef = React.useRef<string>(null);
+
+  const onContentChange = () => {
+    if (modalContentRef.current) {
+      setContentHeight(modalContentRef.current.getBoundingClientRect().height);
+    }
+  };
 
   useEffect(() => {
     setTimeout(() => setPopupTimerComplete(true), popupDelayInMs);
   }, []);
 
   useEffect(() => {
     const containerElement = getContainerElement();
+    const placeholderElement = getPlaceholderElement();
     if (shouldShowModal && !isDomReady) {
       if (!containerElement) {
         const htmlContainer = document.createElement('div');
         htmlContainer.setAttribute('id', containerId);
         htmlContainer.setAttribute('data-testid', 'html-cookie-consent-container');
         document.body.insertBefore(htmlContainer, document.body.firstChild);
       }
+
+      if (!placeholderElement) {
+        const htmlPlaceholder = document.createElement('div');
+        htmlPlaceholder.setAttribute('id', placeholderId);
+        htmlPlaceholder.setAttribute('data-testid', 'html-cookie-consent-placeholder');
+        htmlPlaceholder.setAttribute('aria-hidden', 'true');
+        document.body.lastChild.after(htmlPlaceholder);
+      }
+
       setIsDomReady(true);
     } else if (!shouldShowModal && isDomReady) {
       if (containerElement) {
         document.body.removeChild(containerElement);
       }
+      if (placeholderElement) {
+        document.body.removeChild(placeholderElement);
+      }
       setIsDomReady(false);
     }
   }, [shouldShowModal, isDomReady, setIsDomReady]);
 
+  useEffect(() => {
```

Comment:
This useEffect is probably causing additional re-renderings of the Modal component, also therefore `ReactDOM.createPortal` gets called again. In the main cookie-consent branch, in initial render, the `ReactDOM.createPortal` gets called 2 times. In this branch, it gets called 4 times in the initial render.

Also in the main branch, if you press esc when the modal is open, the `ReactDOM.createPortal` is not called. Neither if you press 'Näytä evästeasetukset'. But in this branch, if you press esc or switch focus out of the modal, then the `ReactDOM.createPortal` will be called again. I wonder if we could reduce the amount of re-renderings? Would combining the `useEffects` help?